core/domain/feedback_jobs_one_off.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for feedback models."""
+from __future__ import absolute_import # pylint: disable=import-only-modules
+from __future__ import unicode_literals # pylint: disable=import-only-modules
+
+from core import jobs
+from core.domain import feedback_services
+from core.domain import user_services
+from core.platform import models
+
+(feedback_models,) = models.Registry.import_models([models.NAMES.feedback])
+
+
+class FeedbackThreadCacheOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to populate message data cache of threads."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """The list of datastore classes to map over."""
+        return [feedback_models.GeneralFeedbackThreadModel]
+
+    @staticmethod
+    def map(thread_model):
+        """Implements the map function for this job."""
+        last_nonempty_message = None
+        for message in reversed(
+                feedback_services.get_messages(thread_model.id)):
+            if message.text:
+                last_nonempty_message = message
+                break
+
+        cache_updated = any([
+            FeedbackThreadCacheOneOffJob._cache_last_nonempty_message_text(
+                thread_model, last_nonempty_message),
```

Comment:
Handle the None case in map() and make sure that what's passed to the other `_cache_*` methods is nonempty, to be in accordance with the typeinfo in the docstring that those methods have. You should then be able to simplify the logic in those methods.

Reply:
Nope -- I think what you've done is fine. Thanks!